lib/pf/Switch.pm

```diff
@@ -3590,8 +3590,9 @@ Set the current tenant in the DAL based on the tenant ID configured in the switc
 =cut
 
 sub setCurrentTenant {
-    my ($self) = @_;
-    pf::dal->set_tenant($self->{_TenantId});
+    my ($self, $radius_request) = @_;
+    my $tenant_id = $radius_request->{"PacketFence-Tenant-Id"} // $self->{_TenantId};
+    pf::dal->set_tenant($tenant_id);
```

Comment:
Why would this be ever different?

Reply:
In my case I was using a RADIUS filter to map the same source switch IP to multiple tenants based on the content of the RADIUS request (like a SmartZone zone name). This PR adds all the plumbing so RADIUS filters can override the tenant ID when needed (both in auth+acct)